tap_stripe/__init__.py

```diff
@@ -884,11 +929,20 @@ def sync():
     for catalog_entry in Context.catalog['streams']:
         stream_name = catalog_entry['tap_stream_id']
         # Sync records for stream
-        if Context.is_selected(stream_name) and not Context.is_sub_stream(stream_name):
-            sync_stream(stream_name)
-            # This prevents us from retrieving 'events.events'
-            if STREAM_TO_TYPE_FILTER.get(stream_name):
-                sync_event_updates(stream_name)
+        if Context.is_selected(stream_name):
+            if not Context.is_sub_stream(stream_name):
+                sync_stream(stream_name)
+                # This prevents us from retrieving 'events.events'
+                if STREAM_TO_TYPE_FILTER.get(stream_name):
+                    bookmark_value = get_bookmark_for_stream(stream_name + '_events', 'updates_created')
+                    sync_event_updates(stream_name, bookmark_value)
+            else:
+                bookmark_value = get_bookmark_for_sub_stream(stream_name)
+                sync_sub_stream(stream_name, bookmark_value)
+                events_bookmark = singer.get_bookmark(Context.state, stream_name + '_events', 'updates_created')
+                if events_bookmark:
+                    bookmark_value = events_bookmark
+                sync_event_updates(stream_name, bookmark_value)
```

Comment:
Add comment above this section about what that section is doing.

Reply:
Added comments